### PR TITLE
Update CLI for redisenterprise 2023-03-01-preview using aaz

### PR DESCRIPTION
**Pull request by revanthballa1188**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
redisenterprise

**Comment by kairu-ms** [comment]
Hi @revanthballa1188, please clean up the commands in generated and manual folders as well. BTW it seems not necessary to update `quantum` extension

**Comment by revanthballa1188** [comment]
@kairu-ms , I did not understand what you said? what do you mean by clean up generated folders?
I will remove the quantum folder.

Also, I need help with removing the database_name as required for the commands, It should be taken automatically from the resource as we have just one default now right now

File: src/demo/azext_demo/custom.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        from azure.cli.core.aaz import AAZStrArg
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.database_name = AAZStrArg(
+            options=["--database-name"],
+            help="The name of the database.",
+            required=False,
+        )
```
[the remark this answers is not in the document]
Comment (kairu-ms) [reply src/demo/azext_demo/custom.py:39]: You should also provide the default name in pre_operations. Like this https://github.com/Azure/azure-cli/blob/0982e1c02d64d3c50d199e83593eeb414133458f/src/azure-cli/azure/cli/command_modules/network/custom.py#L1605-L1611

File: src/redisenterprise/setup.py [hunk 2]
```diff
@@ -1,20 +1,16 @@
...
 from setuptools import setup, find_packages
 
+
 # HISTORY.rst entry.
-VERSION = '0.1.2'
-try:
-    from azext_redisenterprise.manual.version import VERSION
-except ImportError:
-    pass
+VERSION = '0.1.0'
```
Comment (kairu-ms) [inline src/redisenterprise/setup.py:13]: Please use a higher version then previous

File: src/redisenterprise/azext_redisenterprise/azext_metadata.json [hunk 3]
```diff
@@ -1,3 +1,4 @@
 {
-    "azext.minCliCoreVersion": "2.15.0"
+    "azext.isPreview": true,
```
Comment (kairu-ms) [inline src/redisenterprise/azext_redisenterprise/azext_metadata.json:2]: It seems the previous version is not preview. You should remove this line.

File: src/redisenterprise/azext_redisenterprise/__init__.py [hunk 4]
```diff
@@ -1,53 +1,42 @@
...
+# Code generated by aaz-dev-tools
+# --------------------------------------------------------------------------------------------
 
-import azext_redisenterprise._help
 from azure.cli.core import AzCommandsLoader
+from azext_redisenterprise._help import helps  # pylint: disable=unused-import
 
 
-class RedisEnterpriseManagementClientCommandsLoader(AzCommandsLoader):
+class DemoCommandsLoader(AzCommandsLoader):
```
Comment (kairu-ms) [inline src/redisenterprise/azext_redisenterprise/__init__.py:12]: Please rename it to `RedisEnterpriseManagementClientCommandsLoader`

File: src/redisenterprise/azext_redisenterprise/__init__.py [hunk 5]
```diff
@@ -1,53 +1,42 @@
...
-            load_arguments_manual(self, command)
-        except ImportError as e:
-            if e.name.endswith('manual._params'):
-                pass
-            else:
-                raise e
 
 
-COMMAND_LOADER_CLS = RedisEnterpriseManagementClientCommandsLoader
+COMMAND_LOADER_CLS = DemoCommandsLoader
```
Comment (kairu-ms) [inline src/redisenterprise/azext_redisenterprise/__init__.py:42]: Please rename it back here.

Comment (revanthballa1188) [reply src/redisenterprise/azext_redisenterprise/__init__.py]: done

File: src/redisenterprise/azext_redisenterprise/__init__.py (hunk 4 above)
Comment (revanthballa1188) [reply src/redisenterprise/azext_redisenterprise/__init__.py]: done

File: src/redisenterprise/azext_redisenterprise/azext_metadata.json (hunk 3 above)
Comment (revanthballa1188) [reply src/redisenterprise/azext_redisenterprise/azext_metadata.json]: removed

File: src/redisenterprise/setup.py (hunk 2 above)
Comment (revanthballa1188) [reply src/redisenterprise/setup.py]: updated

File: src/demo/azext_demo/custom.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (revanthballa1188) [reply src/demo/azext_demo/custom.py]: done

File: src/redisenterprise/setup.py [hunk 6]
```diff
@@ -1,20 +1,16 @@
...
+# Code generated by aaz-dev-tools
 # --------------------------------------------------------------------------------------------
 
-
 from codecs import open
 from setuptools import setup, find_packages
 
+
 # HISTORY.rst entry.
 VERSION = '0.1.2'
```
Comment (kairu-ms) [inline src/redisenterprise/setup.py:13]: Please bump up version if you want to release it.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**